### PR TITLE
NXP RW6xx: Add additional support to Wakeup from low power modes

### PR DESCRIPTION
**Pull request by mmahadevan108**
This SoC has an external pin interrupt that can be used to wakeup from low power modes.

### CODE REVIEW

File: dts/bindings/power/nxp,aon-wakeup-pin.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    required: true
+    description: Interrupt to wakeup the SoC from lowe power modes.
+
+  wakeup-level:
+    type: int
+    enum:
+      - 0
+      - 1
+    default: 1
+    description: Wakeup level on the pin, default is wakeup on high level.
```
Comment (danieldegrasse) [inline dts/bindings/power/nxp,aon-wakeup-pin.yaml:22]: We need to justify the default here- is this the reset value for the PMU wakeup level? That is usually the reasoning behind a default

**Review by danieldegrasse — COMMENTED** [review]
As this PR stands, it will simply wake the part up- there will be no callback to the application indicating that a level change on pin 24 or 25 caused the interrupt. Personally, I think we should add this support within this PR, but if you'd prefer to enable this more basic functionality in the initial revision I am ok with that as well.

**Comment by mmahadevan108** [comment]
> As this PR stands, it will simply wake the part up- there will be no callback to the application indicating that a level change on pin 24 or 25 caused the interrupt. Personally, I think we should add this support within this PR, but if you'd prefer to enable this more basic functionality in the initial revision I am ok with that as well.

We do not need the notification at this point as we PM mode 3 is not enabled. However I do agree that some form of notification would be required. We should be able to use the GPIO PM suspend and wakeup hooks to notify the GPIO driver. The driver can take appropriate action for pin wakeup.

File: dts/bindings/power/nxp,aon-wakeup-pin.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+compatible: "nxp,aon-wakeup-pin"
+
+include: base.yaml
+
+properties:
+  interrupts:
+    type: array
+    required: true
+    description: Interrupt to wakeup the SoC from lowe power modes.
```
Comment (decsny) [inline dts/bindings/power/nxp,aon-wakeup-pin.yaml:14]: typo

File: soc/nxp/rw/power.c [hunk 3]
```diff
@@ -27,11 +28,36 @@ LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
 
 power_sleep_config_t slp_cfg;
 
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(pin1), okay)
```
Comment (decsny) [inline soc/nxp/rw/power.c:31]: why is there only code for pin1?

Comment (mmahadevan108) [reply soc/nxp/rw/power.c]: Thanks, I have changed this to include pin 0

Comment (decsny) [reply soc/nxp/rw/power.c]: I meant in general in the PR not just this line

Comment (mmahadevan108) [reply soc/nxp/rw/power.c]: I have tested only for Pin 1 with the EVK. We can add Pin 0 once it can be tested.

Comment (decsny) [reply soc/nxp/rw/power.c]: okay

File: dts/bindings/power/nxp,aon-wakeup-pin.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+include: base.yaml
+
+properties:
+  interrupts:
+    type: array
+    required: true
+    description: Interrupt to wakeup the SoC from low power modes.
+
+  wakeup-level:
+    type: int
```
Comment (decsny) [inline dts/bindings/power/nxp,aon-wakeup-pin.yaml:17]: int enum feels awkward, maybe let's do either a boolean or a string enum

File: soc/nxp/rw/power.c [hunk 5]
```diff
@@ -6,6 +6,7 @@
 #include <zephyr/kernel.h>
 #include <zephyr/pm/pm.h>
 #include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
```
Comment (decsny) [inline soc/nxp/rw/power.c:9]: is it really necessary to have a dependency on GPIO to use this feature the way we are enabling it here? I believe all we are needing to do is set one bit to make the pin input? If we do have a dependency on GPIO then we need to capture it in Kconfig

File: dts/arm/nxp/nxp_rw6xx_common.dtsi [hunk 6]
```diff
@@ -147,6 +147,14 @@
 	pmu: pmu@31000 {
 		reg = <0x31000 0x130>;
 		compatible = "nxp,rw-pmu";
+		pin0: pin0 {
```
Comment (decsny) [inline dts/arm/nxp/nxp_rw6xx_common.dtsi:150]: set to status disabled and comment that it is not yet enabled if we don't put code for it

**Review by decsny — CHANGES_REQUESTED** [review]

File: dts/bindings/power/nxp,aon-wakeup-pin.yaml (hunk 4 above)
Comment (mmahadevan108) [reply dts/bindings/power/nxp,aon-wakeup-pin.yaml]: I have changed this to a String.

File: soc/nxp/rw/power.c (hunk 5 above)
Comment (mmahadevan108) [reply soc/nxp/rw/power.c]: I have changed this to use pinctrl directly

File: dts/arm/nxp/nxp_rw6xx_common.dtsi (hunk 6 above)
Comment (mmahadevan108) [reply dts/arm/nxp/nxp_rw6xx_common.dtsi:150]: Done, I have also added support for `pin0` as well and tested with a build locally.

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**